lib/Optimizer/CodeGen/LowerToQIRProfile.cpp

```diff
@@ -207,11 +209,15 @@ struct AddFuncAttribute : public OpRewritePattern<LLVM::LLVMFuncOp> {
     rewriter.startRootUpdate(op);
     const auto &info = iter->second;
     nlohmann::json resultQubitJSON{info.resultQubitVals};
+    std::string profileName = "base_profile";
```

Comment:
```suggestion
    const char * profileName = "base_profile";
```
to avoid allocating heap memory and copying the string literal.

Reply:
Changed in 09a91f29.